src/main/java/it/gov/pagopa/pu/pagopapayments/service/synchronouspayments/SynchronousPaymentService.java

```diff
@@ -86,8 +93,22 @@ public long retrieveNotificationFeeCents(Long organizationId, String nav, String
         return 0;
       }
     } else {
-      //TODO - P4ADEV-2694 if SENDApiKey doesn't exists call external third part API to retrieve notificationFee
-      return 0;
+      try{
+        List<DebtPositionDTO> debtPositionDTOList = debtPositionService.getDebtPositionsByOrganizationIdAndIuv(organizationId,
+          Utilities.nav2Iuv(nav), ORDINARY_DEBT_POSITION_ORIGINS, accessToken);
+        DebtPositionTypeOrg debtPositionTypeOrg = debtPositionService.getDebtPositionTypeOrgById(debtPositionDTOList.getFirst().getDebtPositionTypeOrgId(), accessToken);
+        if(debtPositionTypeOrg.getAmountActualizationOrgSilServiceId()!=null)
+        {
+          log.info("Retrieve notification fee from pu-sil by OrgSilServiceId {} and nav {}", debtPositionTypeOrg.getAmountActualizationOrgSilServiceId(), nav);
+          AmountUpdatesDTO amountUpdatesDTO = puSilService.getAmountUpdates(debtPositionTypeOrg.getAmountActualizationOrgSilServiceId(), nav, accessToken);
+          if (amountUpdatesDTO.getNotificationFee()!=null && amountUpdatesDTO.getNotificationFee()>0)
+            return amountUpdatesDTO.getNotificationFee();
+        }
+      } catch (Exception e) {
+        log.warn("Failed to retrieve notification fee from pu-sil: {}", e.getMessage());
+        return 0;
```

Comment:
the exception NotPayableSilActualizedAmountException should be re-thrown  and translated into Fault code `PAA_DOVUTO_NON_PAGABILE`